staging/src/k8s.io/utils/feature/feature.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package feature provides functions and types for managing feature gates,
+// which are used to control whether specific functionality is enabled.
+package feature
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/spf13/pflag"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+//FIXME: unclear if we want to make the API in terms of "Gates" or "Features".
+//"Features" is so abstract...
```

Comment:
I prefer the more explicit `FeatureGate` and `FeatureGateSet` (or `FeatureSet`)

Reply:
I was trying to follow the "don't stutter" approach.  `feature.Gate` vs `feature.FeatureGate` or heaven forbid `featuregate.FeatureGate`.